clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -784,4 +784,29 @@ public static long computeChecksum(ByteBuffer buffer, int start, int size) {
         return crc.getValue();
     }
 
+    /**
+     * Read data from the channel to the given byte buffer
+     * @param channel File channel containing the data to read from
+     * @param destinationBuffer The buffer into which bytes are to be transferred
+     * @param position The file position at which the transfer is to begin; must be non-negative
+     *
+     * @throws IllegalArgumentException If invalid arguments were passed
+     * @throws IOException If some other I/O error occurs
```

Comment:
@throws IOException If an I/O error occurs. See {@link FileChannel#read(ByteBuffer, long)} for detailed information on the exception cases.